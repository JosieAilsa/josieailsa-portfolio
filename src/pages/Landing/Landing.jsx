 import "./Landing.scss"
 import logo from "../../assets/images/icons/jnlogo-landing2-01.svg"
 import pinkFlower from"../../assets/images/flowers/flowers-01.svg"
 import orangeFlower from "../../assets/images/flowers/flowers-02.svg"
 import whiteFlower from"../../assets/images/flowers/flowers-03.svg"
 import bluePetal from"../../assets/images/flowers/flowers-04.svg"
 import greenLeaf from"../../assets/images/flowers/flowers-05.svg"
 import whitePetals from"../../assets/images/flowers/flowers-06.svg"
 import orangePetals from"../../assets/images/flowers/flowers-07.svg"


const Landing = ({handleLandingHover}) => {



    return (
    <main class="landing" >
        <div className="landing__artwork">
            <img className ="landing__art"src={pinkFlower} alt=""/>
            <img className ="landing__art"src={orangeFlower} alt=""/>
            <img className ="landing__art"src={greenLeaf} alt=""/>
            <img className ="landing__art"src={pinkFlower} alt=""/>
            <img className ="landing__art"src={whiteFlower} alt=""/>
            <img className ="landing__art"src={orangePetals} alt=""/>
            <img className ="landing__art"src={whitePetals} alt=""/>
        </div>
    <section className="landing-hero">
        <img src={logo} alt="jnlogo" srcset="" className="landing-hero__logo"/>
    </section>
        <p onMouseOver={handleLandingHover} className="landing__cta">See portfolio</p>
    </main>
    );
}
 
export default Landing;