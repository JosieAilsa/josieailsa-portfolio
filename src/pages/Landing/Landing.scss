@use "../../assets/sass/abstracts/mixin" as *;
@use "../../assets/sass/abstracts/variables" as *;

.landing {
  width: 100vw;
  height: 100vh;
  background-image: linear-gradient(110deg, #a9bfde 45%, #faf4ed);
  position: relative;
  display: flex;
  justify-content: center;


  .landing__artwork {
    justify-content: space-between;
    width: 100vw;
    height: 100vh;
    display: flex;
    justify-content: space-between;
    flex-wrap: wrap;
    // @media only screen and (min-width: $tablet-breakpoint ){
    //   flex-wrap: nowrap;
    // }
    .landing__art {
      z-index: 1;
      height: 5rem;
      width: 5rem;    
      @media only screen and (min-width: $tablet-breakpoint ){
        height: 10vmin;
        width: 10vmin;
      }


  &:nth-child(1) {
    position: relative;
    animation:  linear infinite;
    animation-name: floatDown;
    animation-delay: random(200)ms;
    animation-duration: 10s;
    top: -10%;
  }
  &:nth-child(2) {
    position: relative;
    animation:  linear infinite;
    animation-name: floatDown;
    animation-delay: 1000ms;
    animation-duration: 15s;
    top: -10%;
  }
  &:nth-child(3) {
    position: relative;
    animation:  linear infinite;
    animation-name: floatDown;
    animation-delay: 0ms;
    animation-duration: 9s;
    top: -10%;
  }
  &:nth-child(4) {
    position: relative;
    animation:  linear infinite;
    animation-name: floatDown;
    animation-delay: 500ms;
    animation-duration: 25s;
    top: -10%;
  }
  &:nth-child(5) {
    position: relative;
    animation:  linear infinite;
    animation-name: floatDown;
    animation-delay: 20ms;
    animation-duration: 17s;
    top: -10%;
  }

  &:nth-child(6) {
    position: relative;
    animation:  linear infinite;
    animation-name: floatDown;
    animation-delay: 1500ms;
    animation-duration: 12s;
    top: -70%;
    @media only screen and (min-width: $tablet-breakpoint ){
      top: -10%;
    }
  }
  &:nth-child(7) {
    position: relative;
    animation:  linear infinite;
    animation-name: floatDown;
    animation-delay: 1000ms;
    animation-duration: 25s;
    top: -50%;
  } 
  &:nth-child(8) {
    position: relative;
    animation:  linear infinite;
    animation-name: floatDown;
    animation-delay: 20ms;
    animation-duration: 12s;
    top: -10%;
  } 
  &:nth-child(9) {
    position: relative;
    animation:  linear infinite;
    animation-name: floatDown;
    animation-delay: 750ms;
    animation-duration: 31s;
    top: -50%;
  } 
  &:nth-child(10) {
    position: relative;
    animation:  linear infinite;
    animation-name: floatDown;
    animation-delay: 2ms;
    animation-duration: 9s;
    top: -50%;
  } 
  &:nth-child(11) {
    position: relative;
    animation:  linear infinite;
    animation-name: floatDown;
    animation-delay: 50ms;
    animation-duration: 7s;
    top: -10%;
  } 
  &:nth-child(12) {
    position: relative;
    animation:  linear infinite;
    animation-name: floatDown;
    animation-delay: 1000ms;
    animation-duration: 7s;
    top: -50%;
  } 
  &:nth-child(13) {
    position: relative;
    animation:  linear infinite;
    animation-name: floatDown;
    animation-delay: 2ms;
    animation-duration: 9s;
    top: -50%;
  } 
  }
}

  .landing-hero {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    z-index: 2;
    height: 60vh;
    position: absolute;
    bottom: 0;
    text-align: center;


    .landing-hero__logo {
      width: 30vmin;
      height: 30vmin;
    }
  }

  .landing__cta {
    padding: 1rem;
    font-family: $font-accent;
    color: $color-green;
    font-size: 1.5rem;
  }
}
@keyframes floatDown{
  0%{
    opacity:0;
    top: -10%;
    transform:translateX(20px) rotate(0deg);
  }
  10%{
    opacity:1;
  }
  20%{
    transform:translateX(-40px) rotate(45deg);
  }
  40%{
    transform:translateX(-20px) rotate(90deg);
  }
  60%{
    transform:translateX(-20px) rotate(180deg);
  }
  80%{
    transform:translateX(-20px) rotate(200deg);
  }
  100%{
    top: 110%;
    transform:translateX(-20px) rotate(225deg);
    opacity:0;
  }
}

@keyframes slideInLeft {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 100;
  }
}