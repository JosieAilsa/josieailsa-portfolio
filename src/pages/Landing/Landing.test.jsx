import Landing from "./Landing"
import { customRender } from "../../utils/testutils";

test("The landing page", () => {
    const { container } = customRender(<Landing/>);
    expect(container).toMatchSnapshot();
  });
  