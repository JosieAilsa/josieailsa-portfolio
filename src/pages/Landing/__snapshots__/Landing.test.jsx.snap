// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The landing page 1`] = `
<div>
  <main
    class="landing"
  >
    <div
      class="landing__artwork"
    >
      <img
        alt=""
        class="landing__art landing__flower-1"
        src="flowers-01.svg"
      />
      <img
        alt=""
        class="landing__art landing__flower-2"
        src="flowers-02.svg"
      />
      <img
        alt=""
        class="landing__art landing__flower-3"
        src="flowers-05.svg"
      />
      <img
        alt=""
        class="landing__art landing__flower-4"
        src="flowers-01.svg"
      />
      <img
        alt=""
        class="landing__art landing__flower-5"
        src="flowers-03.svg"
      />
      <img
        alt=""
        class="landing__art landing__flower-6"
        src="flowers-07.svg"
      />
      <img
        alt=""
        class="landing__art landing__flower-7"
        src="flowers-06.svg"
      />
    </div>
    <section
      class="landing-hero"
    >
      <img
        alt="jnlogo"
        class="landing-hero__logo"
        src="jnlogo-landing2-01.svg"
        srcset=""
      />
    </section>
    <p
      class="landing__cta"
    >
      See portfolio
    </p>
  </main>
</div>
`;
